files/scripts/check-rebase.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+
+WARNING_MSG = (
+    "Your branch is not up to date with upstream/master. \n"
+    "SHA of the last commit of upstream/master: {upstream}\n"
+    "Please, rebase!"
+)
+
+
+def main():
+    path = os.path.dirname(os.path.abspath(__file__))
+
+    last_commit_subject = subprocess.run(
+        ["git", "log", "--format=%s", "-1"], capture_output=True, cwd=path
+    ).stdout.decode()
+    print(f"Last commit subject: {last_commit_subject}")
+
+    if "Merge commit" in last_commit_subject:
+        print("Zuul merged the master -- rebase is needed.")
```

Comment:
You can use some more verbose message as you have in `WARNING_MSG`. (I don't mean it to be the final version..;-)